api/v1/submissions/PKPSubmissionHandler.php

```diff
@@ -550,6 +556,40 @@ public function delete($slimRequest, $response, $args)
         return $response->withJson($submissionProps, 200);
     }
 
+    /**
+     * Get the decisions associated to a submission
+     *
+     * @param Request $slimRequest Slim request object
+     * @param Response $response object
+     * @param array $args arguments
+     *
+     * @return Response
+     */
+    public function getDecisions($slimRequest, $response, $args)
+    {
+        $request = Application::get()->getRequest();
+        $context = $request->getContext();
+        $submission = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION);
+
+        if (!$submission || $submission->getData('contextId') !== $context->getId()) {
+            return $response->withStatus(404)->withJsonError('api.404.resourceNotFound');
+        }
+
+        $data = [];
+
+        $decisionIterator = Repo::decision()->getCollector()
+            ->filterBySubmissionIds([$submission->getId()])
+            ->getMany();
+        
+        $map = Repo::decision()->getSchemaMap();
+
+        foreach ($decisionIterator as $decision) {
+            $data[] = $map->map($decision);
+        }
```

Comment:
There should be a `mapMany()` function that you can use: `Repo::decision()->getSchemaMap()->mapMany($decisionIterator)`. When you pass the JSON data back, you'll want ot reset the array keys so:

```php
$data = Repo::decision()
    ->getSchemaMap()
    ->mapMany($decisionIterator->values());
```

Reply:
Done.